include/aws/common/array_list.inl

```diff
@@ -86,7 +86,26 @@ bool aws_array_list_is_valid(const struct aws_array_list *AWS_RESTRICT list) {
     bool required_size_is_valid = (aws_mul_size_checked(list->length, list->item_size, &required_size) == AWS_OP_SUCCESS);
     bool current_size_is_valid = (list->current_size >= required_size);
     bool data_is_valid = ((list->current_size == 0 && list->data == NULL) || AWS_MEM_IS_WRITABLE(list->data, list->current_size));
-    return required_size_is_valid && current_size_is_valid && data_is_valid;
+    bool item_size_is_valid = (list->item_size!=0);
+    return required_size_is_valid && current_size_is_valid && data_is_valid && item_size_is_valid;}
+
+AWS_STATIC_IMPL
+bool aws_array_list_is_wiped(const struct aws_array_list *AWS_RESTRICT list) {
+    if(!list) {
+        return false;
+    }
+    bool current_size_is_wiped = (list->current_size == 0);
+    bool item_size_is_wiped = (list->item_size == 0);
+    bool length_is_wiped = (list->length == 0);
+    bool data_is_wiped = (list->data == NULL);
+    bool alloc_is_wiped = (list->alloc == NULL);
+    return current_size_is_wiped && item_size_is_wiped && length_is_wiped && data_is_wiped && alloc_is_wiped;
+}
+
+AWS_STATIC_IMPL
+void aws_array_list_debug_print(const struct aws_array_list * list) {
```

Comment:
Maybe this (and other debug print functions) should go in a different file debug.c or something.

Reply:
Yeah if there are other debug print functions we could do that, but I think that would be separate from this PR.